Modules/Core/Mesh/include/itkTriangleMeshCurvatureCalculator.h

```diff
@@ -0,0 +1,166 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+/*=========================================================================
+ *
+ *  Portions of this file are subject to the VTK Toolkit Version 3 copyright.
+ *
+ *  Copyright (c) Ken Martin, Will Schroeder, Bill Lorensen
+ *
+ *  For complete copyright, license and disclaimer of warranty information
+ *  please refer to the NOTICE file at the top of the ITK source tree.
+ *
+ *=========================================================================*/
+
+#ifndef itkTriangleMeshCurvatureCalculator_h
+#define itkTriangleMeshCurvatureCalculator_h
+
+#include <iostream>
+#include "itkMesh.h"
+
+namespace itk
+{
+/**  \class TriangleMeshCurvatureCalculator
+ * \brief
+ * Calculator to compute curvature of a triangle mesh. Set the input triangle mesh and the
+ * required curvature type first. Default curvature type is Gauss. After computing curvature the result
+ * can be obtained using the getter method. It throws exeption if the input mesh is not set.
+ * The implementation is same as in VTK.
+ * \ingroup ITKMesh
+ */
+
+template <typename TInputMesh>
+class ITK_TEMPLATE_EXPORT TriangleMeshCurvatureCalculator : public Object
+{
+public:
+  ITK_DISALLOW_COPY_AND_MOVE(TriangleMeshCurvatureCalculator);
+
+  /** Standard class type aliases. */
+  using Self = TriangleMeshCurvatureCalculator;
+
+  /** Standard "Superclass" type alias. */
+  using Superclass = Object;
+
+  /** Smart pointer type alias support. */
+  using Pointer = SmartPointer<Self>;
+  using ConstPointer = SmartPointer<const Self>;
+
+  /** Method for creation through the object factory. */
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(TriangleMeshCurvatureCalculator, Object);
+
+  /** Convenient type alias for this Calculator. */
+  using InputMeshType = TInputMesh;
+  using MeshConstPointer = typename InputMeshType::ConstPointer;
+  using MeshPointType = typename InputMeshType::PointType;
+  using MeshPointIdConstIterator = typename InputMeshType::CellTraits::PointIdConstIterator;
+  using MeshPointsContainerConstPointer = typename InputMeshType::PointsContainerConstPointer;
+  using MeshPointDataContainer = typename InputMeshType::PointDataContainer;
+  using MeshPointDataContainerPointer = typename InputMeshType::PointDataContainerPointer;
+  using CellsContainerConstPointer = typename InputMeshType::CellsContainerConstPointer;
+  using CellsContainerConstIterator = typename InputMeshType::CellsContainer::ConstIterator;
+  using CellType = typename InputMeshType::CellType;
+
+  /** \class CurvatureEnum
+   * \brief Different modes of operation. Currently only Guassian curvature is supported.
+   * Using enum for future extension to Mean, Max and Min curvature.
+   */
+  enum class CurvatureEnum : uint8_t
```

Comment:
Strongly typed enums are re declared  outside the class as per the ITK migration guide:
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/ITK5MigrationGuide.md#strongly-typed-enumerations

And they should also be wrapped.

Please exercise the ostream overaload in the test.

Reply:
The ostream overload needs to go into a cxx implementation file, yes. Sorry for the oversight.

Reply:
I did the enum wrapping and changed the container for m_GaussCurvatureData to double so that Mesh with custom data types will also work. Earlier I created the container same as the Mesh's point data container which can differ as per the specification.

Reply:
This comment fits best elsewhere, maybe as a separate comment in the PR, as it is unrelated to the enum-related thread (?), and if it is important it might go unnoticed.

Reply:
Adding it in commit message.